Move the `tsh` config file guidance

ptgott opened the pull request:
Fixes #27817

Move our guidance on the `tsh` config files to the `tsh` guide in the "Connect your Client" section and out of the CLI reference. This is because the CLI reference focuses on commands, while the general `tsh` guide covers a number of topics, making it a more appropriate place for the config file guide.

zmb3 commented:
Can this be backported beyond 13? We have some upcoming changes on the way that will need to be backported, so having the changes land in the same file on all 3 branches is preferred.

ptgott commented:
@zmb3 I'll need to backport the CLI reference split to v11 and v12 to ensure this happens smoothly. I'll make time for that this week.

lsgunn-teleport commented on docs/pages/connect-your-client/tsh.mdx:
```diff
@@ -896,11 +896,193 @@ exit
+
+`tsh example 0 1 unused-2 3 unused-4` will expand to `bash -c 'echo first=0 0-1 3 unused-2 unused-4'`.
+
+An alias definition can also reference the `$TSH` variable. `tsh` will expand this variable to the absolute path of the current `tsh` executable. This can be useful if there are multiple `tsh` versions installed, or the currently used version is not in `PATH`.
```
```suggestion
An alias definition can also reference the `$TSH` variable. If you use the $TSH variable in an alias, `tsh` expands the variable to the absolute path of the current `tsh` executable. This behavior can be useful if there are multiple `tsh` versions installed, or the version you're currently using is not in the `PATH`.
```

lsgunn-teleport commented on docs/pages/connect-your-client/tsh.mdx:
````diff
@@ -896,11 +896,193 @@ exit
+    "status": "$TSH status --format=json"
+```
+
+To troubleshoot aliases, set the `TELEPORT_DEBUG=1` environment variable. This will cause detailed logs to be printed to the console:
````
```suggestion
To troubleshoot aliases, set the `TELEPORT_DEBUG=1` environment variable. With this variable, detailed logs are displayed as standard output. For example:
```

ptgott replied:
👍 ,though this should be "standard error". I'm also not sure what "With this variable" adds, so I've preserved "This".

lsgunn-teleport commented on docs/pages/connect-your-client/tsh.mdx:
```diff
@@ -896,11 +896,193 @@ exit
+
+### Proxy Templates
+
+With Proxy Templates, `tsh` will dynamically determine the address of the
+Proxy Service to connect to based on the address of the destination host
+in your `tsh ssh` or `tsh proxy ssh` command.
```
```suggestion
With proxy templates, `tsh` dynamically determines the address of the
Teleport Proxy Service to connect to based on the address of the destination host
in your `tsh ssh` or `tsh proxy ssh` command.
```

lsgunn-teleport commented on docs/pages/connect-your-client/tsh.mdx:
```diff
@@ -896,11 +896,193 @@ exit
+host server address `%h:%p` with the configured templates. For each replace rule set,
+the corresponding cli value will be set.
+
+If leaf certificates are required to connect to the node, `tsh` will automatically
+retrieve leaf certificates from the root cluster.
```
```suggestion
If leaf certificates are required to connect to the node, `tsh` automatically
retrieves leaf certificates from the root cluster.
```

lsgunn-teleport commented on docs/pages/connect-your-client/tsh.mdx:
```diff
@@ -896,11 +896,193 @@ exit
+  host: "$1:$3"
+
+# Multiple templates can be provided. They are evaluated in order and the first
+# one matching will take effect.
```
```suggestion
# match takes effect.
```

lsgunn-teleport commented on docs/pages/connect-your-client/tsh.mdx:
````diff
@@ -896,11 +896,193 @@ exit
+ERROR: proxy jump contains {{proxy}} variable but did not match any of the templates in tsh config
+```
+
+If the proxy variable `-J {{proxy}}` is not explicitly provided, then `tsh` will
+still attempt to match a template, but will not fail when there is no match.
+Additionally, it will not replace the `proxy` value when explicitly set by the
+client.
````
```suggestion
If you don't explicitly provide the proxy variable `-J {{proxy}}`, `tsh`
still attempts to match a template, but won't fail if there isn't a match.
Additionally, `tsh` won't replace the `proxy` value if it's explicitly set by the
client.
```

lsgunn-teleport reviewed (approved):
A few suggestions for active voice, present tense and such

ptgott committed:
Move the `tsh` config file guidance

Fixes #27817

Move our guidance on the `tsh` config files to the `tsh` guide in the
"Connect your Client" section and out of the CLI reference. This is
because the CLI reference focuses on commands, while the general `tsh`
guide covers a number of topics, making it a more appropriate place for
the config file guide.
